### PR TITLE
Get by name func

### LINKED ISSUE #52: getByName from database
This issue involves 3 parts:
1. create a new route: 
     - GET /recipe/searchByTitle/:title
2. (optional) refactor countRecipes and use the new route to get the list of recipes, then resolve it with it's length
3. refactor getByName to work as follows
```
/**
 * TODO: make a new route for this function
 * @param {Array} recipeNames A list of recipe name queries.
 * @returns {Promise} Resolves with the number of recipes matching the given names,
 *                    rejects if it fails.
 */
async function countRecipes(recipeNames) {
  return new Promise((resolve) => {
    const numRecipes = 0;
    const namesProcessed = 0;
    if (recipeNames.length === 0) {
      resolve(0);
      return;
    }
    recipeNames.forEach((recipeName) => {
      // TODO: replace below with fetch call.
      // db.recipes.where('recipe_name').equals(recipeName).count()
      //   .then((count) => {
      //     numRecipes += count;
      //     namesProcessed += 1;
      //     if (namesProcessed === recipeNames.length) {
      //       resolve(numRecipes);
      //     }
      //   });
    });
  });
}
```
```
/**
 * TODO: need a route for this as well. GET /recipes/:title
 * @param {String} queryName The recipe name query.
 * @returns {Promise} Resolves with an array of recipe JSONs whose name contains the query,
 *                    rejects if it fails.
 */
async function getByName(query) {
  return new Promise((resolve, reject) => {
    if (typeof query !== 'string') {
      reject(new Error('Query was not a string!'));
    } else {
      const recipesPushed = 0;
      const recipeNames = [];
      let filteredNames;
      const queryLower = query.toLowerCase();
      // TODO: modify the below stuff
      // db.recipes.orderBy('recipe_name').eachUniqueKey((name) => {
      //   recipeNames.push(name);
      // })
      //   .then(() => {
      //     filteredNames = recipeNames.filter((name) => name.toLowerCase().includes(queryLower));
      //     return countRecipes(filteredNames);
      //   })
      //   .then((numRecipes) => {
      //     const jsonArray = [];
      //     if (numRecipes === 0) {
      //       resolve(jsonArray);
      //       return;
      //     }
      //     filteredNames.forEach(async (name) => {
      //       const collection = db.recipes.where('recipe_name').equals(name);
      //       collection.each((recipe) => {
      //         jsonArray.push(recipe.recipe_data);
      //         recipesPushed += 1;
      //       })
      //         .then(() => {
      //           if (recipesPushed === numRecipes) {
      //             resolve(jsonArray);
      //           }
      //         });
      //     });
      //   });
    }
  });
}
```

### PR DESCRIPTION
**Pull request by demyinn00**
# Resolves #52 
After creating your PR, please complete this template!

### What were your changes
- Created a new route and model for the route:
     - getByTitle in /source/backend/database/models/recipes.js
     - GET /recipes/searchByTitle/:title in /source/backend/routes/recipes.js
- I modified the getByName function inside /source/frontend/assets/scripts/database.js
I have commented out the countRecipes function because I'm sure it's unnecessary and I will completely remove it after it's been approved. 

### What specifically should be checked
- Test if this works. I had trouble testing so I'm not sure if the site works with that search. I am confident the route works because I was able to test that on postman.

### CODE REVIEW

File: source/frontend/assets/scripts/database.js [hunk 1]
```diff
@@ -133,27 +133,25 @@ async function getBySpice(spiceLevel) {
...
-      //     numRecipes += count;
-      //     namesProcessed += 1;
-      //     if (namesProcessed === recipeNames.length) {
-      //       resolve(numRecipes);
-      //     }
-      //   });
-    });
-  });
-}
+// async function countRecipes(recipeNames) {
```
Comment (mingkaili) [inline source/frontend/assets/scripts/database.js:136]: We can safely delete lines 136 - 154

**Review by mingkaili — CHANGES_REQUESTED** [review]

File: source/backend/database/models/recipesModel.js [hunk 2]
```diff
@@ -180,6 +181,13 @@ async function getBySpiceRating(spiceRating) {
   return result;
 }
 
+async function getByTitle(title) {
+  const result = await db('recipes')
+    .select('*')
+    .where({ title });
```
[the remark this answers is not in the document]
Comment (demyinn00) [reply source/backend/database/models/recipesModel.js:187]: Should the `title` be title case or should it be lowercase? Inside the database everything seems to be title case, or does `whereRaw` ignore that? I looked into `whereRaw` and `whereLike` and it seems it is case sensitive, so I am assuming it needs to be title case?

File: source/frontend/assets/scripts/database.js [hunk 3]
```diff
@@ -166,10 +164,22 @@ async function getByName(query) {
     if (typeof query !== 'string') {
       reject(new Error('Query was not a string!'));
     } else {
-      const recipesPushed = 0;
-      const recipeNames = [];
-      let filteredNames;
       const queryLower = query.toLowerCase();
+      try {
+        const recipesArray = await fetch(`${url}/recipes/searchByTitle/${recipeName}`);
```
[the remark this answers is not in the document]
Comment (demyinn00) [reply source/frontend/assets/scripts/database.js:169]: Yes that makes perfect sense. I completely forgot that async basically does everything a promise can, but it's more readable. Thank you

[the remark this answers is not in the document]
Comment (demyinn00) [reply source/frontend/assets/scripts/database.js]: Fixed!

**Pull request merged**